homeassistant/components/tradfri/switch.py

```diff
@@ -36,14 +51,14 @@ def _refresh(self, device):
         self._device_data = device.socket_control.sockets[0]
 
     @property
-    def is_on(self):
+    def is_on(self) -> bool:
         """Return true if switch is on."""
-        return self._device_data.state
+        return cast(bool, self._device_data.state)
```

Comment:
Any reason why you prefer `cast()` over `bool()` here? For me, `return bool(self._device_data.state)` feels more natural.

Reply:
cast() does not affect the code, it is more like an instruction to pyc and mypy, it does not change anything.

bool() on the other hand actually interpret the variable and delivers a changed result.

Reply:
That's right, but since your method claims to return `bool`, it should actually return `bool` (IMHO).

Reply:
I'm not sure what the library returns here, but if we need to change it to return a boolean I think we can copy it to a boolean here with `bool`.

Reply:
When I looked at the library it did return bool, but it was not clear if other types could be returned as well. For that reason I prefer a cast() which is non-intrusive. Apart from I just realized that there are a very small runtime fee for using bool, since runtime will call a function, whereas cast is like a macro (have a look at the pyc file, with bool() and cast() respectively).

Reply:
In reality I would like to make the library typed, wdyt ?

Reply:
Typed librares are (IMHO) always better than untyped libraries.